tests/unit/testaddonsessionstate.cpp

```diff
@@ -0,0 +1,97 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "testaddonsessionstate.h"
+
+#include "addons/state/addonsessionstate.h"
+
+void TestAddonSessionState::testGetAndSet() {
+  QJsonObject numProp;
+  numProp["type"] = "number";
+  numProp["default"] = 10;
+  QJsonObject manifest;
+  manifest["numProp"] = numProp;
+
+  AddonSessionState* state = AddonSessionState::fromManifest(manifest);
```

Comment:
Without a parent you leak this object.

Reply:
Yes that is why I had the explicit `delete`... But I'll change that.